src/nnet3/nnet-optimize.cc

```diff
@@ -879,6 +903,7 @@ const NnetComputation* CachingOptimizingCompiler::Compile(
     }
     computation_.ComputeCudaIndexes();
   }
+  UpdateCache();
```

Comment:
The whole point of this exercise is to avoid redundantly compiling things we have compiled before- for speed.  Think about this more carefully.